### PR TITLE
Ticketing system new

### PR DESCRIPTION
**Pull request by Bharani0012**
Added Comments API - Ticket_Service.py

Added Comments Functions - db_manager.py

### CODE REVIEW

File: src/backend/db/database_manager.py [hunk 1]
```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+        except Exception as e:
+            logger.error(f"Unexpected error while creating comment: {e}")
+            session.rollback()
+            raise e
+
+        finally:
+            logger.debug("Closing database session")
+            session.close()
+
+    def get_comment_by_id(self, comment_id, customer_guid):
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:970]: Change function to have four parameters def get_comment_by_id(self, comment_id, ticket_id, customer_guid):

File: src/backend/db/database_manager.py [hunk 2]
```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Retrieve comment details
+            comment_query = '''
+            SELECT comment_id, ticket_id, posted_by, comment, created_at, is_edited
+                FROM ticket_comments
+                WHERE comment_id = :comment_id
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:991]: add and where ticket_id = ticket_id

File: src/backend/db/database_manager.py [hunk 3]
```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+        except (OperationalError, DatabaseError) as e:
+            logger.error(f"Database error: {e}")
+            raise Exception("Database connectivity issue")
+        except SQLAlchemyError as e:
+            logger.error(f"SQLAlchemy error: {e}")
+            raise Exception("Database query error")
+        finally:
+            session.close()
+
+    def get_comments_by_ticket_id(self, customer_guid, ticket_id):
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:1019]: Make it a paginated get API

Check for getallchats how it is done

File: src/backend/db/database_manager.py [hunk 4]
```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                        "SELECT updated_at, is_edited FROM ticket_comments WHERE ticket_id = :ticket_id AND comment_id = :comment_id"),
+                    {"ticket_id": ticket_id, "comment_id": comment_id}
+                ).fetchone()
+
+                session.commit()
+
+                updated_at = updated_at_result[0] if updated_at_result else None
+                is_edited = updated_at_result[1] if updated_at_result else False
+
+                return {"status": "updated", "reason": None, "is_edited":is_edited, "updated_at": updated_at}
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:1135]: return the updated comments {comment: comment, ticket_id: ticket_id and comment_id:comment_id} also in the return field

File: src/backend/ticket_service/ticket_service.py [hunk 5]
```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+            logger.error(f"Database error: {e}")
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal Server Error")
+
+
+@app.put("/tickets/{ticket_id}/comments/{comment_id}", response_model=CommentUpdateResponse, tags=["Comment Management"])
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:528]: This api is not intuitive

Make it /updatecomment

And then add the ticket id and comment_id in the parameters

File: src/backend/ticket_service/ticket_service.py [hunk 6]
```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+    except Exception as e:
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="An unexpected server error occurred."
+        )
+
+
+@app.delete("/tickets/{ticket_id}/comments/{comment_id}", response_model=CommentResponse, tags=["Comment Management"])
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:588]: This API is again not intuitive
Make it /deletecomment

And then add the ticket id and comment_id in the parameters

File: src/backend/ticket_service/ticket_service.py [hunk 7]
```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+
+    except Exception as e:
+        logger.error(f"Unexpected error in creating comment: {e}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            detail="An unexpected error occurred"
+        )
+
+
+@app.get("/comments/{comment_id}", response_model=Comment, tags=["Comment Management"])
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:464]: make it /tickets/ticket_id/comments/comment_id

File: src/backend/ticket_service/ticket_service.py [hunk 8]
```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+class CommentUpdate(BaseModel):
+    comment: str
+
+class CommentUpdateResponse(BaseModel):
+    comment_id: str
+    comment: str
+    is_edited: bool
+    updated_at: datetime
+
+@app.post("/comments", response_model=CommentResponse, status_code=HTTPStatus.CREATED, tags=["Comment Management"])
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:418]: Make it /addcomment

File: src/backend/ticket_service/ticket_service.py [hunk 9]
```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+        logger.debug("Calling DBManager.create_comment")
+        db_response = db_manager.create_comment(
+            comment.customer_guid,
+            comment.ticket_id,
+            comment.posted_by,
+            comment.comment
+        )
+
+        logger.debug(f"Database response: {db_response}")
+        return CommentResponse(comment_id=db_response["comment_id"], status="created")
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:434]: Return the comment_id and the comment
Don't return the status field

**Review by SaravanakumarR2018 — COMMENTED** [review]
You can add the positive and negative testcases file like what you did for ticketing system

File: src/backend/db/database_manager.py (hunk 1 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:970]: Modified Sir

File: src/backend/db/database_manager.py (hunk 2 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:991]: Modified Sir

File: src/backend/ticket_service/ticket_service.py (hunk 9 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:434]: Modified Sir

File: src/backend/ticket_service/ticket_service.py (hunk 8 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:418]: modified sir

File: src/backend/ticket_service/ticket_service.py (hunk 7 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:464]: modified sir

File: src/backend/ticket_service/ticket_service.py (hunk 5 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:528]: modified sir

File: src/backend/db/database_manager.py (hunk 4 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:1135]: modified sir

File: src/backend/ticket_service/ticket_service.py (hunk 6 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:588]: modified sir

File: src/backend/db/database_manager.py (hunk 3 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:1019]: modified sir

**Pull request closed without merging**